docs/hoon/demo.md

```diff
@@ -24,65 +25,60 @@ line is commented with its number:
     "Fizz"                                              ::  11
   ?:  =(0 (mod count 3))                                ::  12
     "Buzz"                                              ::  13
-  (scow %ud count)                                      ::  14
+  <count>                                               ::  14
 $(count (add 1 count))                                  ::  15
 ```
 
-Hoon is a functional language.  FizzBuzz is usually defined as a
-side effect, but officially Hoon has no side effects.  Accordingly,
-this code will *produce* a list of strings, not *print* a list of
-strings. (Of course, if you run this code from the Urbit
-[_Dojo_](../../using/shell) (shell) you will be able to see the
-product.)
+When FizzBuzz is written in non-functional languages the 
+output is generated as a side effect of the program, e.g., with 
+`printf()`.  Hoon is a functional language so technically it 
+has no side effects.  Accordingly, the code above *returns* a 
+list of strings, it doesn't print them. (If you run it from the 
+Urbit shell, [_Dojo_](../../using/shell), you will of course be 
+able to see the product.)
```

Comment:
*returns/produces nitpicking intensifies*

Reply:
Tbh I'm not clear on the anti-"return" crusade.  What's actually wrong with it?  Functions return values; it's common, intuitive language.

Reply:
Produces is more functional.

Reply:
That's just not true.  The word "returns" is ubiquitous and far more common than "produces" in discussions of functions.

https://en.wikipedia.org/wiki/Function_(mathematics)
https://en.wikipedia.org/wiki/Function_composition_%28computer_science%29

Reply:
It could be that "return" suggests the possibility of the absence of a return (as in void functions in an imperative language). Honestly, I've been crusading so long I don't even remember.

@cgyarvin?

Reply:
The only reason I'm pushing on this (it's not that big a deal really) is that lots of people are coming from elsewhere having used the word "return" (either in math or software dev).  If we scrupulously avoid the word "return", it suggests that "produce" means something importantly different, when as far as I can tell it doesn't.  If there's some technical reason to avoid it, fine, but technical reasons don't apply as much to this early Hoon outreach doc, which should have intuitive language.